pytorch_tabnet/multitask.py

```diff
@@ -291,33 +131,81 @@ def predict(self, X):
 
         Returns
         -------
-            predictions: np.array
-                Predictions of the most probable class
+            batch_outs: dict
         """
         self.network.eval()
-        dataloader = DataLoader(PredictDataset(X),
-                                batch_size=self.batch_size, shuffle=False)
+
+        dataloader = DataLoader(
+            PredictDataset(X), batch_size=self.batch_size, shuffle=False
+        )
 
         results = {}
         for data in dataloader:
             data = data.to(self.device).float()
             output, _ = self.network(data)
-            predictions = [torch.argmax(torch.nn.Softmax(dim=1)(task_output),
-                                        dim=1).cpu().detach().numpy().reshape(-1)
-                           for task_output in output]
-
+            predictions = [
+                torch.nn.Softmax(dim=1)(task_output).cpu().detach().numpy()
+                for task_output in output
+            ]
             for task_idx in range(len(self.output_dim)):
                 results[task_idx] = results.get(task_idx, []) + [predictions[task_idx]]
-        # stack all task individually
-        results = [np.hstack(task_res) for task_res in results.values()]
-        # map all task individually
-        results = [np.vectorize(self.preds_mapper[task_idx].get)(task_res)
-                   for task_idx, task_res in enumerate(results)]
-        return results
+        res = [np.vstack(task_res) for task_res in results.values()]
+        return res
 
-    def predict_proba(self, X):
+
+class TabNetMultiTaskRegressor(TabModel):
```

Comment:
not ok with that, looping over tasks creates some unwanted side effects - list as output, foor loop over tasks etc.. These unwanted side effects can't be handled easily for multi task classification as you don't have the same number of classes for each task. For regression it's different, and doing Multi task regression comes for free thanks to final MLP layer. I does not make sense to create a breaking change that actually makes things slower and more difficult to use.